Framework/Geometry/test/InstrumentTest.h

```diff
@@ -546,26 +546,88 @@ class InstrumentTestPerformance : public CxxTest::TestSuite {
+    access_instrument(*m_instrumentNotParameterized, 1, 16);
+  }
 
-    const detid_t nPixels = 100 * 100 * 6;
-    double pos_x = 0;
-    for (detid_t i = 1; i <= nPixels; i++) {
-      pos_x += m_instrumentNotParameterized->getDetector(i)->getPos().X();
-    }
+  void test_access_non_parameterized_2_threads() {
+    access_instrument(*m_instrumentNotParameterized, 2, 16);
   }
 
-  void test_access_parameterized() {
+  void test_access_non_parameterized_4_threads() {
+    access_instrument(*m_instrumentNotParameterized, 4, 16);
+  }
 
-    const detid_t nPixels = 100 * 100 * 6;
-    double pos_x = 0;
-    for (detid_t i = 1; i <= nPixels; i++) {
-      pos_x += m_instrumentParameterized->getDetector(i)->getPos().X();
-    }
+  void test_access_non_parameterized_8_threads() {
+    access_instrument(*m_instrumentNotParameterized, 8, 16);
+  }
+
+  void test_access_non_parameterized_12_threads() {
+    access_instrument(*m_instrumentNotParameterized, 12, 16);
+  }
+
+  void test_access_non_parameterized_16_threads() {
+    access_instrument(*m_instrumentNotParameterized, 16, 16);
+  }
+
+  void test_access_non_parameterized_20_threads() {
+    access_instrument(*m_instrumentNotParameterized, 20, 16);
+  }
+
+  void test_access_non_parameterized_24_threads() {
+    access_instrument(*m_instrumentNotParameterized, 24, 16);
+  }
+
+  void test_access_parameterized_1_threads() {
+    access_instrument(*m_instrumentParameterized, 1, 16);
+  }
+
+  void test_access_parameterized_2_threads() {
+    access_instrument(*m_instrumentParameterized, 2, 16);
   }
 
+  void test_access_parameterized_4_threads() {
+    access_instrument(*m_instrumentParameterized, 4, 16);
+  }
+
+  void test_access_parameterized_8_threads() {
+    access_instrument(*m_instrumentParameterized, 8, 16);
+  }
+
+  void test_access_parameterized_12_threads() {
+    access_instrument(*m_instrumentParameterized, 12, 16);
+  }
+
+  void test_access_parameterized_16_threads() {
+    access_instrument(*m_instrumentParameterized, 16, 16);
+  }
+
+  void test_access_parameterized_20_threads() {
+    access_instrument(*m_instrumentParameterized, 20, 16);
+  }
+
+  void test_access_parameterized_24_threads() {
+    access_instrument(*m_instrumentParameterized, 24, 16);
+  }
 
 private:
+  void access_instrument(const Instrument &instrument, int numberOfThreads,
+                         int repetitions = 1) {
+    auto threads = std::min(numberOfThreads, omp_get_max_threads());
+
+    const detid_t nPixels = 100 * 100 * 6;
+    double sum = 0.0;
+    for (int rep = 0; rep < repetitions; ++rep) {
+      std::vector<double> pos_x(threads, 0.0);
+#pragma omp parallel for num_threads(threads)
+      for (detid_t i = 1; i <= nPixels; i++) {
+        int thread = omp_get_thread_num();
+        pos_x[thread] += instrument.getDetector(i)->getPos().X();
+      }
+      sum += std::accumulate(begin(pos_x), end(pos_x), 0, std::plus<double>());
+    }
+  }
+
```

Comment:
As I check the performance tests weekly, I'd be able to figure out this had happened, this wouldn't be the only performance test affected by an upgrade